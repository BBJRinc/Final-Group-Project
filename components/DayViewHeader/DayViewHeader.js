import React, { Component } from 'react'
import { View, Image, StyleSheet, ImageBackground, TouchableHighlight } from "react-native";
import { Container, Separator, Content, Icon, Header, ListIcon, Footer, Button, Text, List, Left, Right, Body, ListItem, } from 'native-base';
import moment from 'moment';
import IconF from 'react-native-vector-icons/Feather';
import IconEV from 'react-native-vector-icons/EvilIcons';
import axios from 'axios';


export default class DayViewHeader extends Component {
    constructor(props) {
        super(props)

        this.state = {
            currentDate: ''
        }
    }

    componentDidMount() {
        let today = moment().format("MMM Do YY")
        this.setState({
            currentDate: today
        })
    }

    render() {
        const { nextDay, previousDay } = this.props
        return (
            <Container>

                <Header>

                    <Left Icon>
                        <IconF name="chevron-left"
                            color="#000000"
                            size={15}
                            onPress={() => getPreviousDay()}
                        />
                    </Left>

                    <Body>
                        <Text>
                            {this.state.currentDate}
                        </Text>
                    </Body>

                    <Right>
                        <IconF
                            name="chevron-right"
                            color="#000000"
                            size={15}
                            onPress={() => nextDay()}
                        />
                    </Right>

                </Header>

            </Container>
        )
    }
}

const styles = StyleSheet.create({
    header: {
        height: 10,
        width: '100%',
        backgroundColor: '#ffffff',
    },
    text:{
        color: "#000000"
    }

})