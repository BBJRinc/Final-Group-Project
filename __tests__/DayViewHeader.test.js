import 'react-native';
import React from 'react';
import DayViewHeader from '../components/DayViewHeader/DayViewHeader';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <DayViewHeader />
  ).toJSON();
  expect(tree).toMatchSnapshot();;
});

// Ross Test