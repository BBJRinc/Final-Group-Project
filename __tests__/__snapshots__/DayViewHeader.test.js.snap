// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 15,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, .85)",
          "height": 35,
          "justifyContent": "center",
        },
      ]
    }
    virtual={undefined}
  >
    <View
      Icon={true}
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "flex": 1,
            "paddingRight": 0,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 28,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        юдй
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "flex": 1,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        
      </Text>
    </View>
    <View
      Icon={true}
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          },
          Object {
            "flex": 1,
            "paddingLeft": 0,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 28,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        юдк
      </Text>
    </View>
  </View>
</View>
`;
